packages/components/src/query-controls/stories/index.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import QueryControls from '..';
+import type { Category, QueryControlsProps } from '../types';
+
+const meta: ComponentMeta< typeof QueryControls > = {
+	title: 'Components/QueryControls',
+	component: QueryControls,
+	parameters: {
+		actions: { argTypesRegex: '^on.*' },
+		controls: { expanded: true },
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const DefaultTemplate: ComponentStory< typeof QueryControls > = ( props ) => {
+	const [ ownNumberOfItems, setOwnNumberOfItems ] = useState(
+		props.numberOfItems
+	);
+	const [ ownOrder, setOwnOrder ] = useState( props.order );
+	const [ ownOrderBy, setOwnOrderBy ] = useState( props.orderBy );
+	const [ ownSelectedAuthorId, setOwnSelectedAuthorId ] = useState(
+		props.selectedAuthorId
+	);
+	const [ ownSelectedCategories, setOwnSelectedCategories ] = useState(
+		props.selectedCategories
+	);
+
+	const onCategoryChange: QueryControlsProps[ 'onCategoryChange' ] = (
+		tokens
+	) => {
+		if ( 'string' === typeof tokens ) {
+			return;
+		}
+
+		const allCategories = tokens?.map( ( token ) => {
+			return typeof token === 'string'
+				? props.categorySuggestions?.[ token ]
+				: token;
+		} ) as Category[];
+
+		setOwnSelectedCategories( allCategories );
```

Comment:
Mainly [taken from](https://github.com/WordPress/gutenberg/blob/96a8df71832b07453dca716d0e93094d97db5357/packages/block-library/src/latest-posts/edit.js#L182-L203) the 'Latest Posts' block

Reply:
I proposed a few tweaks to this function as part of the previous comment

Reply:
Thanks